docs/deploy_narrative.md

```diff
@@ -1,64 +1,109 @@
+#### Release CI
 
 Once a pull request is merged from a temporary branch to `develop` a new _development_ image will be available at:
 `docker.pkg.github.com/kbase/narrative/narrative-develop:latest`
 
-Simply restart the `narrative` container for CI in Rancher. No other action is needed.
+To see your changes once the new image is created, simply open a narrative on CI. If you already have one loaded, restart the application by selecting the "Shutdown and Restart" option from the menu in the upper left.
+
+### Stage a Final Release
+
+1.  Update versions following (roughly) [semantic versioning](https://semver.org). The files to be updated are:
+    -   `package.json`, `package-lock.json`, `bower.json`, `src/config.json`, `src/config.json.templ`, `src/biokbase/narrative/\_\_init\_\_.py`
+    -   Major - backward incompatible changes, like the move from Python 2 -> Python 3, or Narrative typed object changes. These are generally considered to be changes that have a strong impact on the Narrative Interface and are hard or impossible to move backward from.
+    -   Minor - new features that don't affect compatibility.
+    -   Patch - adjustments to existing features, bug fixes.  
+
+2.  Add release notes in `RELEASE_NOTES.md`
+    -   Add a new heading with the updated version.
+    -   Add a list of changes with some reasonable detail. You don't need to add the name of every module changed, just the overall feature affected. Include JIRA ticket numbers where appropriate.
 
-### Appdev, Next, & Prod
+3.  PR these changes to the develop branch.
+    -   Ensure that tests pass.
 
-#### Create Image
+4.  PR the develop branch to the master branch.
+    -   Deploy on the next environment.
+    -   Ensure that things work as expected.
 
-1. Create a new [pull request](https://github.com/kbase/narrative/compare) to merge the `develop` branch into `master`.
-1. Follow the same process outlined in the `Develop` section above.
+5.  Create a new release in the narrative repo.
+    -   tag it with the new version, prefixed with v (e.g. v4.2.1).
+
+### Deploying Next, Appdev, & Prod
+
+These environments: <https://next.kbase.us>, <https://appdev.kbase.us>, and <https://narrative.kbase.us> are all deployed using a production image. They're used for slightly different purposes:
+
+-   **next** is used for final internal testing before being promoted to the public-facing environments - appdev and production. Most services should be up to date here, and this is used for final integration and acceptance testing.
+-   **appdev** is used primarily for KBase app development using the [KBase SDK](https://kbase.github.io/kb_sdk_docs/). This environment mirrors the production environment, but allows for finding and running apps in "dev" mode.
+-   **prod** (or narrative.kbase.us) is the main production environment. This is the place where most users will use KBase and run apps that have either been entirely released or have been made available for beta testing.
```

Comment:
This could go at the top, in a "terminology" or "glossary" type section -- so that the names and uses of the environments are made clear before you talk about deployment to those environments.

Reply:
Yup, good call. I moved those to the top and added an entry for CI.